src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -0,0 +1,334 @@
+    //     **Tab**".
+    //   * "P" would return "Close **P**ane", "[-] S**p**lit Horizontal", "[ | ]
+    //     S**p**lit Vertical", "**P**rev Tab", "O**p**en Settings" and "O**p**en Media
+    //     Controls".
+    //   * "sv" would return "[ | ] Split Vertical" (by matching the **S** in
+    //     "Split", then the **V** in "Vertical").
+    // Arguments:
+    // - searchText: the string of text to search for in `name`
+    // - name: the name to check
+    // Return Value:
+    // - true if name contained all the characters of searchText
+    bool CommandPalette::_filterMatchesName(const winrt::hstring& searchText, const winrt::hstring& name)
+    {
+        std::wstring lowercaseSearchText{ searchText.c_str() };
+        std::wstring lowercaseName{ name.c_str() };
+        std::transform(lowercaseSearchText.begin(), lowercaseSearchText.end(), lowercaseSearchText.begin(), std::towlower);
+        std::transform(lowercaseName.begin(), lowercaseName.end(), lowercaseName.begin(), std::towlower);
+
+        const wchar_t* namePtr = lowercaseName.c_str();
+        const wchar_t* endOfName = lowercaseName.c_str() + lowercaseName.size();
+        for (const auto& wch : lowercaseSearchText)
+        {
+            while (wch != *namePtr)
+            {
+                // increment the character to look at in the name string
+                namePtr++;
+                if (namePtr >= endOfName)
+                {
+                    // If we are at the end of the name string, we haven't found all the search chars
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    // Function Description:
+    // - Calculates a "weighting" by which should be used to order a command
+    //   name relative to other names, given a specific search string.
+    //   Currently, this is based off of two factors:
+    //   * The weight is incremented once for each matched character of the
+    //     search text.
+    //   * If a matching character from the search text was found at the start
+    //     of a word in the name, then we increment the weight again.
+    //     * For example, for a search string "sp", we want "Split Pane" to
+    //       appear in the list before "Close Pane"
+    // Arguments:
+    // - searchText: the string of text to search for in `name`
+    // - name: the name to check
+    // Return Value:
+    // - the relative weight of this match
+    int CommandPalette::_getWeight(const winrt::hstring& searchText,
+                                   const winrt::hstring& name)
+    {
+        std::wstring lowercaseSearchText{ searchText.c_str() };
+        std::wstring lowercaseName{ name.c_str() };
+        std::transform(lowercaseSearchText.begin(), lowercaseSearchText.end(), lowercaseSearchText.begin(), std::towlower);
+        std::transform(lowercaseName.begin(), lowercaseName.end(), lowercaseName.begin(), std::towlower);
```

Comment:
You might not even need to create a temporary variable for the lower case text because it looks like you always access each character once so you could put the `std::tolower`s directly in the loop, like:

```c++
int CommandPalette::_getWeight(const winrt::hstring& searchText, const winrt::hstring& name)
{
    int weight = 0;

    auto it = name.cbegin();
    for (wchar_t searchC : searchText)
    {
        searchC = std::towlower(searchC);

        // Advance the iterator to the next character that we're looking
        // for.
        bool lastWasSpace = true;
        while (true)
        {
            // If we are at the end of the name string, we haven't found
            // it.
            if (it == name.cend())
            {
                return false;
            }

            // found it
            if (std::towlower(*it) == searchC)
            {
                break;
            }

            lastWasSpace = *it == L' ';

            ++it;
        }

        // Advance the iterator by one character so that we don't
        // end up on the same character in the next iteration.
        ++it;

        weight += lastWasSpace ? 2 : 1;
    }

    return weight;
}
```

(Note: I used iterators because of the comment above.)